tests/infrastructure/golden_images/update_boot_source/conftest.py

```diff
@@ -125,3 +125,33 @@ def custom_data_source_scope_function(admin_client, custom_data_import_cron_scop
             f"{custom_data_import_cron_scope_function.namespace} namespace."
         )
         raise
+
+
+@pytest.fixture()
+def boot_source_data_from_data_source_dict(auto_update_data_source_matrix__function__):
+    return auto_update_data_source_matrix__function__[[*auto_update_data_source_matrix__function__][0]]
+
+
+@pytest.fixture()
+def existing_data_source_volume(
+    golden_images_persistent_volume_claims_scope_function,
+    golden_images_volume_snapshot_scope_function,
+    matrix_data_source,
+):
+    source = matrix_data_source.source
+    if source.kind == PersistentVolumeClaim.kind:
+        cluster_volumes = golden_images_persistent_volume_claims_scope_function
+    else:
+        cluster_volumes = golden_images_volume_snapshot_scope_function
+    assert any([source.name in volume.name for volume in cluster_volumes]), (
+        f"DataSource source {source.kind} {source.name} is missing"
+    )
+    return matrix_data_source
```

Comment:
Why do you need `source`? We need to use `sourceRef` when we clone from a DataSource, so we wouldn't  need to worry about DataSource implementation, here is an example yaml:

```
apiVersion: cdi.kubevirt.io/v1beta1
kind: DataVolume
metadata:
  name: dv-clone-from-source
spec:
  sourceRef:
    kind: DataSource
    name: rhel8
    namespace: openshift-virtualization-os-images
  storage:
    resources: {}
    storageClassName: ocs-storagecluster-ceph-rbd-virtualization
```

Reply:
I didn't understand what you mean.
This sections checks if the DataSource has an existing source, a snapshot or a PVC.
Do you mean we should check if the source exists in another way? if so can you please suggest a code example?